pybop/models/base_model.py

```diff
@@ -238,37 +246,36 @@ def set_initial_state(self, initial_state: dict):
         # Use a copy of the updated default parameter set
         self._parameter_set = self._unprocessed_parameter_set.copy()
 
-    def set_params(self, rebuild: bool = False, dataset: Dataset = None):
+    def set_current_function(self, dataset: Union[Dataset, dict]):
```

Comment:
Perhaps this should be a private method, since it wouldn't trigger a rebuild.

Reply:
Please can you expand- when would this require a rebuild?

Reply:
Yeah no worries, here's the interaction I'm thinking of:

If I've built the model, then call this method, will the `_built_model` have the updated `_parameter_set` from this method? I don't think it would, but I may be missing the code flow here. If it wouldn't have the updated `_parameter_set` object, I think we want this method to be private: `_set_current_function`, so users know they shouldn't expect it to work correctly with a single call.

Reply:
Thanks, I've updated and added a test

Reply:
Just to note, this works for `model.predict`, but won't for `model.simulate`. Anyway, something we can fix in the future if it pops up.

Reply:
I've just added model.simulate to the test. Please can you explain when it would not work?

Reply:
Seems to work, so I will go ahead and merge